lib/screens/signup/signup.screen.dart

```diff
@@ -0,0 +1,236 @@
+  TextEditingController _passController = new TextEditingController();
+  TextEditingController _nameController = new TextEditingController();
+  final _formkey = GlobalKey<FormState>();
+
+  registration() async {
+    if (password != null&& _nameController.text!=""&& _emailController.text!="") {
+      try {
+        UserCredential userCredential = await FirebaseAuth.instance
+            .createUserWithEmailAndPassword(email: email, password: password);
+        ScaffoldMessenger.of(context).showSnackBar(SnackBar(
+            content: Text(
+              "Registered Successfully",
+              style: TextStyle(fontSize: 20.0),
+            )));
+        // ignore: use_build_context_synchronously
+        Navigator.push(
+            context, MaterialPageRoute(builder: (context) => HomeScreen()));
+      } on FirebaseAuthException catch (e) {
+        if (e.code == 'weak-password') {
+          ScaffoldMessenger.of(context).showSnackBar(SnackBar(
+              backgroundColor: Colors.orangeAccent,
+              content: Text(
+                "Password Provided is too Weak",
+                style: TextStyle(fontSize: 18.0),
+              )));
+        } else if (e.code == "email-already-in-use") {
+          ScaffoldMessenger.of(context).showSnackBar(SnackBar(
+              backgroundColor: Colors.orangeAccent,
+              content: Text(
+                "Account Already exists",
+                style: TextStyle(fontSize: 18.0),
+              )));
+        }
+      }
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+
+
+
+    return SafeArea(
+      child: Scaffold(
+        body: Container(
+          decoration: const BoxDecoration(
+            image: DecorationImage(
+              image: AssetImage("assets/SL-051919-20420-09.jpg"),
+              fit: BoxFit.cover,
+            ),
+          ),
+          child: Container(
+            margin: const EdgeInsets.all(24),
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+              children: [
+                _header(context),
+                _inputField(context),
+                _loginin(context),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  _header(BuildContext context) {
+    return Row(
+      children: [
+        Image.asset(
+          "assets/logo.png", // Replace with your image asset path
+          height: 90, // Adjust the height as needed
+        ),
+        const SizedBox(width: 10), // Add some space between the logo and text
+        Expanded(
+          child: LayoutBuilder(
+            builder: (context, constraints) {
+              final maxWidth = constraints.maxWidth;
+              final textStyle = TextStyle(
+                fontSize: 30.0,
+                fontWeight: FontWeight.bold,
+                color: Color(0xFF36A383),
+              );
+
+              final welcomeText = Text(
```

Comment:
tot legat de header, vad ca il folositi peste tot in astea 3 ecrane :) daca aveti chef ati putea sa faceti un widget pe care sa il afisati peste tot, in loc sa il creati in fiecare ecran in parte ;)